vault.py

```diff
@@ -74,22 +74,30 @@ def run(self, terms, inject=None, variables=None, **kwargs):
             raise AnsibleError('Vault authentication token missing. Specify with'
                                ' VAULT_TOKEN environment variable or $HOME/.vault-token')
 
+        cafile = os.getenv('VAULT_CACERT')
+        capath = os.getenv('VAULT_CAPATH')
         try:
-            context = ssl.create_default_context(cafile=os.getenv('VAULT_CACERT'),
-                                                 capath=os.getenv('VAULT_CAPATH'))
+            if cafile or capath:
+                context = ssl.create_default_context(cafile=cafile, capath=capath)
+            else:
+                context = None
             request_url = urljoin(url, "v1/%s" % (key))
             req = urllib2.Request(request_url, data)
             req.add_header('X-Vault-Token', token)
             req.add_header('Content-Type', 'application/json')
-            response = urllib2.urlopen(req, context=context)
+            if context:
```

Comment:
I see your objection and what you're trying to do.

Just to clarify, however, while passing `context=None` works in 2.7.9:

```
$ python
Python 2.7.9 (default, Nov  3 2016, 15:11:53) 
[GCC 4.8.4] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import urllib2
>>> req = urllib2.Request("http://google.com")
>>> urllib2.urlopen(req, context=None)
<addinfourl at 139925302741328 whose fp = <socket._fileobject object at 0x7f42ea4f7850>>
```

it fails in 2.7.8:

```
$ python
Python 2.7.8 (default, Nov  2 2016, 17:29:37) 
[GCC 4.8.4] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import urllib2
>>> req = urllib2.Request("http://google.com")
>>> urllib2.urlopen(req, context=None)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: urlopen() got an unexpected keyword argument 'context'
>>>
```

I know you don't want to jump through a lot of hoops to support Python < 2.7.9. But I feel the compromise I'm offering here (see https://github.com/jhaals/ansible-vault/pull/26/commits/b73fb191846ad32d7612936b6c7f822d6131349f) is less cluttered, more Pythonic, and a small price to pay to continuing to work on Ubuntu 14.04 LTS.